src/concat.js

```diff
@@ -25,12 +25,12 @@ var _isArray = require('./internal/_isArray');
  *      R.concat([4, 5, 6], [1, 2, 3]); //=> [4, 5, 6, 1, 2, 3]
  *      R.concat('ABC', 'DEF'); // 'ABCDEF'
  */
-module.exports = _curry2(function(set1, set2) {
-  if (_isArray(set2)) {
-    return _concat(set1, set2);
-  } else if (_hasMethod('concat', set1)) {
-    return set1.concat(set2);
+module.exports = _curry2(function(a, b) {
+  if (_isArray(a) && _isArray(b)) {
+    return _concat(a, b);
+  } else if (_hasMethod('concat', a)) {
```

Comment:
What mechanism would you use to check too that 

    concat(AddInt(4), AddNum(Math.PI))

should throw an error, but

    concat(AddNum(Math.PI), AddInt(4))

is perfectly legitimate because `AddInt` is a subclass of `AddNum` but not the reverse?